### PR TITLE
Add streaks feature

### LINKED ISSUE #100: Add streaks panel
As a user, I can see how many consecutive goals I achieved so that I would be more motivated to maintain my streak.

### PR DESCRIPTION
**Pull request by ivantjh**
- Add a new `Streaks` details panel
- Change the goal tracking to be more forgiving (i.e. `weekly`'s deadline is on the following Sunday of `lastMeetingDate` plus one week)
- `monthly`'s deadline changed to last day of each month and `yearly`'s deadline changed to last day of each year
- Need wait for #91 to be up to start using. You can still manually switch to the new pane in `DetailsPanel.java`

Fix #100

### CODE REVIEW

**Review by lue97 — APPROVED** [review]
I'm not sure if it's better to be more 'lenient'. For instance If I were to set a yearly deadline and meet on 1st Jan then the deadline will be approximately 2 year away.

Code wise LGTM

**Comment by ivantjh** [comment]
Yeap, that's correct. I think you can think of it as weekly homework. If I were to complete my weekly homework on Monday when it is supposed to be due on Friday, I don't want the following week's homework deadline to be shifted to next Monday.

File: src/main/java/seedu/address/ui/DetailsPanel.java [hunk 1]
```diff
@@ -7,38 +7,49 @@
...
 
     /**
      * Creates a {@code DetailsPanel} with the given {@code ObservableList}s.
+     *
      * @param upcomingEventsList A list of upcoming events.
-     * @param detailedPerson A list containing a single {@code Person}.
+     * @param detailedPerson     A list containing a single {@code Person}.
      */
-    public DetailsPanel(ObservableList<PersonEvent> upcomingEventsList, ObservableList<Person> detailedPerson) {
+    public DetailsPanel(ObservableList<PersonEvent> upcomingEventsList,
```
Comment (chesterhow) [inline src/main/java/seedu/address/ui/DetailsPanel.java:29]: Remember to update the documentation to include the new `personStreaks` param

File: src/test/java/seedu/address/testutil/EventBuilder.java [hunk 2]
```diff
@@ -48,7 +48,14 @@ public EventBuilder withDescription(String description) {
         return this;
     }
 
+    /**
+     * Builds an Event with the attributes from EventBuilder
+     */
     public Event build() {
+        if (time == null) {
+            return new Event(date, description);
+        }
```
Comment (chesterhow) [inline src/test/java/seedu/address/testutil/EventBuilder.java]: I don't think you need to handle this case anymore as I've removed the `time` param from the `requireAllNonNull` check in `Event`'s constructor

**Review by chesterhow — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/ui/DetailsPanel.java (hunk 1 above)
Comment (ivantjh) [reply src/main/java/seedu/address/ui/DetailsPanel.java:29]: Will add as an issue first!

**Pull request merged**